scout/models/omics_variant.py

```diff
@@ -0,0 +1,189 @@
+""" OMICS variant
+
+    For potentially causative variants that are not yet in ClinVar
+    and have yet not been marked causative in any existing case.
+
+"""
+
+import logging
+from datetime import datetime
+from typing import List, Optional
+
+from pydantic import BaseModel, Field, field_validator, model_validator
+
+LOG = logging.getLogger(__name__)
+
+
+class OmicsVariantLoader(BaseModel):
+    """Omics variants loader
+    OmicsVariants are e.g. RNA expression outliers as identified by the DROP pipeline.
+
+    Variable names are as found in the original files, plus a set common to all mixed in after file parsing,
+    but before model validation by this class.
+
+    The serialisation names will be used when dumping the model for e.g. db storage.
+    """
+
+    case_id: str
+    institute: str
+    build: str = "38"
+    variant_type: str = "clinical"
+    category: str = "outlier"
+    sub_category: str = "splicing"
+    date: datetime = datetime.now()
+    display_name: str
+    omics_variant_id: str
+    # omics variant id hash (including clinical/research)?
+
+    # DROP Fraser and Outrider outlier TSVs
+
+    # sample id is mandatory: each row pertains to one outlier event in one individual as compared to others
+    # In the db object, this will be replaced with a "samples" array of individual dict.
+    sampleID: str
+
+    # outlier variants must identify the gene they pertain to, primarily with an hgnc_id
+    hgnc_ids: Optional[List[int]] = Field(alias="hgnc_id", serialization_alias="hgnc_ids")
+    geneID: Optional[str]
+
+    hgnc_symbols: Optional[List[str]] = Field(
+        alias="hgncSymbol", serialization_alias="hgnc_symbols"
+    )
+    gene_name_orig: Optional[str]
+
+    gene_type: Optional[str]
+
+    # coordinates if applicable
+    chromosome: Optional[str] = Field(alias="seqnames", serialization_alias="chromosome")
+    position: Optional[int] = Field(alias="start", serialization_alias="position")
+    end: Optional[int]
+    width: Optional[int] = None
+    strand: Optional[str] = None
+
+    pValue: Optional[float]
+
+    # Fraser specific
+    type: Optional[str] = None
+    psiValue: Optional[float] = None
+    deltaPsi: Optional[float] = None
+    counts: Optional[int] = None
+    totalCounts: Optional[int] = None
+    meanCounts: Optional[float] = None
+    meanTotalCounts: Optional[float] = None
+    nonsplitCounts: Optional[int] = None
+    nonsplitProportion: Optional[float] = None
+    nonsplitProportion_99quantile: Optional[float] = None
+    annotatedJunction: Optional[str] = None
+    pValueGene: Optional[float] = None
+    padjustGene: Optional[float] = None
+    PAIRED_END: Optional[str] = None
+    isExternal: Optional[bool] = None
+    potentialImpact: Optional[str] = None
+    causesFrameshift: Optional[str] = None
+    UTR_overlap: Optional[str] = None
+    blacklist: Optional[bool] = None
```

Comment:
We don't really test for `blacklist`. Maybe we just skip it until it's rena..actually in use.